pkg/network/server.go

```diff
@@ -1252,7 +1252,7 @@ func (s *Server) requestBlocks(bq Blockqueuer, p Peer) error {
 	lq := s.bQueue.lastQueued()
 	if lq > pl.IndexStart {
 		c := int16(h + blockCacheSize - lq)
-		if c < payload.MaxHashesCount {
+		if c < payload.MaxHashesCount && c > 0 {
```

Comment:
This is not correct, if we have `c <= 0` then we're likely in a situation where some time has passed between `bq.BlockHeight()` and `s.bQueue.lastQueued()`, so we have an old height (0, for example) with some new last queued (4000, for example, 2000 blocks consumed already). This means that we should be doing a bit different thing here, we need both `lq` and the capacity left in the queue (`blockCacheSize - bq.len`) at the same time returned from the block queue. If the capacity is zero, we can't do a lot and no package should be sent at all, if it's less than `MaxHashesCount` then `pl.Count` should be adjusted for it.

Reply:
> the only possible cause is zero (or <-1, but it's probably not the case)

Just for information, we've checked it, payload with `0` blocks count was requested in this case:
```
дек 15 17:44:32 az neo-go[998229]: 2022-12-15T17:44:32.814Z        WARN        peer disconnected        {"addr": "10.78.69.115:55634", "error": "invalid block count: 0", "peerCount": 2}
```

Reply:
Fixed.